File: backend/app/schemas/missa.py
from datetime import date, datetime
from typing import Optional

from pydantic import BaseModel


class BlocoResponse(BaseModel):
    id: int
    ordem: int
    tipo: str
    titulo: Optional[str]
    referencia: Optional[str]
    conteudo: Optional[str]
    conteudo_formatado: Optional[str] = None
    visivel: bool

    class Config:
        from_attributes = True


class MissaResponse(BaseModel):
    id: int
    data: date
    celebracao: Optional[str]
    subtitulo: Optional[str] = None
    descricao: Optional[str] = None
    tempo_liturgico: Optional[str]
    status_processamento: str
    total_blocos: int = 0

    class Config:
        from_attributes = True


class MissaListResponse(BaseModel):
    id: int
    data: date
    celebracao: Optional[str]
    status_processamento: str


class MissaCompletaResponse(BaseModel):
    id: int
    data: date
    celebracao: Optional[str]
    subtitulo: Optional[str] = None
    descricao: Optional[str] = None
    tempo_liturgico: Optional[str]
    fonte_pdf_url: str
    pdf_hash: Optional[str]
    status_processamento: str
    data_criacao: datetime
    blocos: list[BlocoResponse]

    class Config:
        from_attributes = True

File: web/src/types/missa.ts
export interface PalavraDoDia {
  texto: string
  referencia: string
}

export interface Missa {
  id: number
  data: string
  celebracao: string | null
  subtitulo: string | null
  descricao: string | null
  tempo_liturgico: string | null
  status_processamento: string
  total_blocos: number
  palavra_do_dia?: PalavraDoDia | null
}

export interface BlocoLiturgico {
  id: number
  ordem: number
  tipo: string
  titulo: string | null
  referencia: string | null
  conteudo: string | null
  visivel: boolean
}

export interface MissaCompleta {
  id: number
  data: string
  celebracao: string | null
  tempo_liturgico: string | null
  fonte_pdf_url: string
  pdf_hash: string | null
  status_processamento: string
  data_criacao: string
  blocos: BlocoLiturgico[]
}
